common/src/main/java/org/apache/sedona/common/utils/BBox.java

```diff
@@ -38,6 +37,64 @@ public BBox(double startLon, double endLon, double startLat, double endLat) {
     this.endLat = endLat;
   }
 
+  public BBox(Geometry geom) {
+    startLon = Double.MAX_VALUE;
+    endLon = -Double.MAX_VALUE;
+    startLat = Double.MAX_VALUE;
+    endLat = -Double.MAX_VALUE;
+    minZ = Double.MAX_VALUE;
+    maxZ = -Double.MAX_VALUE;
+    processEnvelope(geom);
+  }
+
+  private void processEnvelope(Geometry geom) {
```

Comment:
Why is this so complicated? You can see how existing approach read the coordinates https://github.com/apache/sedona/blob/master/common/src/main/java/org/apache/sedona/common/Functions.java#L495